src/components/VueTypeaheadBootstrapListItem.vue

```diff
@@ -40,15 +40,29 @@ export default {
     backgroundVariant: {
       type: String
     },
+    backgroundVariantResolver: {
+      type: Function,
+      default: (d) => null,
+      validator: d => d instanceof Function
+    },
     textVariant: {
       type: String
     }
   },
+  data: function() {
```

Comment:
I pulled these static classes out of textClasses() and put them into data - makes it easier to possibly inject these values in the future, and makes it easier to run some test scenarios.